catalog/core/catalog-core-contentresourcereader/src/test/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProviderTest.java

```diff
@@ -0,0 +1,148 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+
+public class DerivedContentActionProviderTest {
+
+    private static DerivedContentActionProvider actionProvider;
+
+    private static ActionProvider mockResourceActionProvider = mock(ActionProvider.class);
+
+    private static final String CONTENT_ID = UUID.randomUUID()
+            .toString();
+
+    private static Metacard metacard = mock(Metacard.class);
+
+    private static Attribute attribute = mock(Attribute.class);
+
+    private static URI actionUri;
+
+    private static URI derivedResourceUri;
+
+    private static final String QUALIFIER = "qualifier";
+
+    @BeforeClass
+    public static void init() throws URISyntaxException {
+        actionUri = new URI("https://example.com/download?transform=resource");
+        derivedResourceUri = new URI(ContentItem.CONTENT_SCHEME, CONTENT_ID, QUALIFIER);
+        actionProvider = new DerivedContentActionProvider(mockResourceActionProvider);
+
+    }
+
+    @Before
+    public void setUp() {
+        when(metacard.getId()).thenReturn(CONTENT_ID);
+        when(metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI)).thenReturn(attribute);
+        when(attribute.getValues()).thenReturn(Arrays.asList(derivedResourceUri.toString()));
+    }
+
+    @Test
+    public void testGetActions() throws Exception {
+
+        ActionImpl expectedAction = new ActionImpl("expected",
+                "expected",
+                "expected",
+                actionUri.toURL());
+        when(mockResourceActionProvider.getActions(any(Metacard.class))).thenReturn(Arrays.asList(
+                expectedAction));
+        List<Action> actions = actionProvider.getActions(metacard);
+        assertThat(actions, hasSize(1));
+        assertThat(actions.get(0), notNullValue());
+        assertThat(actions.get(0)
+                .getUrl(), notNullValue());
+        assertThat(actions.get(0)
+                .getUrl()
+                .getQuery(), containsString("options=" + QUALIFIER));
+
+    }
+
+    @Test
+    public void testGetActionsNonContentUri() throws Exception {
+        ActionImpl expectedAction = new ActionImpl("expected",
+                "expected",
+                "expected",
+                actionUri.toURL());
+        when(mockResourceActionProvider.getActions(any(Metacard.class))).thenReturn(Arrays.asList(
+                expectedAction));
+        when(attribute.getValues()).thenReturn(Arrays.asList("https://exmaple.com/other"));
```

Comment:
make the url string its own variable so it's not repeated below, and change Arrays.asList to Collections.singletonList